lib/takeaway.rb

```diff
@@ -0,0 +1,48 @@
+require_relative 'order'
+require_relative 'menu'
+require_relative 'message'
+
+class Takeaway
+
+  attr_reader :menu, :order
+
+  def initialize(menu = Menu.new, message = Message.new)
+    @message = message
+    @menu = menu
+    @order = nil
+  end
+
+  def display_menu
+    menu.display
+  end
+
+  def order(food, quantity=1)
+    fail 'There is no such dish on the menu' unless on_menu?(food)
+    @order = Order.new(@menu.dish_list) if @order.nil?
+    @order.add_to_order(food, quantity)
+  end
+
+  def basket_summary
+    @order.nil? ? "Basket empty" : @order.basket
+  end
+
+  def total
+    "Total: £#{@order.current_total}"
+  end
+
+  def confirm_order
+    fail 'No orders have been added' if @order.nil?
+    @message.send_text if price_is_correct?
+  end
+
+  private
+
+  def price_is_correct?
+    @order.current_total == @order.total_price_of_basket
```

Comment:
I'm not sure I like this. It feels like you have two sources of truth for the basket price: one is calculated on-the-fly (`#total_price_of_basket`) and the other is stored on the object (`current_total`). What are the cases where they might not match?